### PR TITLE
Build Scopes Configuration

### PR DESCRIPTION
**Pull request by wpopielarski**
There are 3 build scopes defined: macros, main and tests. Up to now
there was a convention how source folders were assigned to particular
scope. This commit adds a configuration which allows Scala IDE user to
assign the particular source folder to required scope via 'Scopes
Settings' tab in 'Scala Compiler' preference page.
Known limitation: when project's 'Properties' is open and user adds
new source folder in 'Java Build Path -> Source' the 'Properties'
popup window must be reopened to make visible newly added source path
in 'Scala Compiler -> Scope Settings'.

### CODE REVIEW

File: org.scala-ide.sdt.core/src/org/scalaide/ui/internal/preferences/ScopesSettings.scala [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        ChoiceSetting(srcName, helpArg = EmptyString, descr = EmptyString, choices, findDefaultScope(srcFolderRelativeToProject))
+      }
+    }.getOrElse(Nil).toList.sortBy { _.name }
+  }
+
+  def makeKey(srcFolderRelativeToProject: IPath): String =
+    NamePrefix + srcFolderRelativeToProject.segments.mkString("/")
+
+  private def findDefaultScope(srcFolderRelativeToProject: IPath): String =
+    CompileScope.scopesInCompileOrder.find { _.isValidSourcePath(srcFolderRelativeToProject) }.get.name
```
Comment (sschaef) [inline org.scala-ide.sdt.core/src/org/scalaide/ui/internal/preferences/ScopesSettings.scala]: Is that `get` safe here?

**Comment by sschaef** [comment]
Source code looks code but again, it would be nice if changes regarding formatting would be moved to their own commit.

**Comment by wpopielarski** [comment]
but this is just autoformatter

2015-09-02 17:10 GMT+02:00 Simon Schäfer <notifications@github.com>:

File: org.scala-ide.sdt.core/src/org/scalaide/ui/internal/preferences/ScopesSettings.scala (hunk 1 above)
Comment (wpopielarski) [inline org.scala-ide.sdt.core/src/org/scalaide/ui/internal/preferences/ScopesSettings.scala]: must not be the case when none of scopes is chosen

**Comment by sschaef** [comment]
LGTM

**Pull request merged**